orchagent/fdborch.cpp

```diff
@@ -174,41 +173,155 @@ bool FdbOrch::storeFdbEntryState(const FdbUpdate& update)
+            
+            if (itr->first.mac == mac || mac == flush_mac)
+            {
+                bool found = m_portsOrch->getPort(itr->first.port_name, port);
+                port_temp = found ?  port : port_empty; 
+                updates.emplace_back(itr->first.mac, itr->first.bv_id, fdb_type, port_temp, false);
+            }
+        }
+    }
+    else if (bv_id == SAI_NULL_OBJECT_ID)
+    {
+        /* FLUSH based on PORT */
+        for (auto itr = m_entries.begin(); itr != m_entries.end(); itr++)
+        {
+            if (itr->second.type.find(fdb_type) == std::string::npos) continue;
+            if (itr->second.bridge_port_id == bridge_port_id)
+            {
+                if (itr->first.mac == mac || mac == flush_mac)
+                {
+                    bool found = m_portsOrch->getPort(itr->first.port_name, port);
+                    port_temp = found ?  port : port_empty; 
+                    updates.emplace_back(itr->first.mac, itr->first.bv_id, fdb_type, port_temp, false);
+                }
+            }  
+        }
+    }
+    else if (bridge_port_id == SAI_NULL_OBJECT_ID)
+    {
+        /* FLUSH based on BV_ID */
+        for (auto itr = m_entries.begin(); itr != m_entries.end(); itr++)
+        {
+            if (itr->second.type.find(fdb_type) == std::string::npos) continue;
+            if (itr->first.bv_id == bv_id)
+            {
+                if (itr->first.mac == mac || mac == flush_mac)
+                {
+                    bool found = m_portsOrch->getPort(itr->first.port_name, port);
+                    port_temp = found ?  port : port_empty;
+                    updates.emplace_back(itr->first.mac, itr->first.bv_id, fdb_type, port_temp, false);
+                }
+            }
+        }
+    }
+    else
+    {
+        /* FLUSH based on port and VLAN */
+        for (auto itr = m_entries.begin(); itr != m_entries.end(); itr++)
+        {
+            if (itr->second.type.find(fdb_type) == std::string::npos) continue;
+            if (itr->first.bv_id == bv_id && itr->second.bridge_port_id == bridge_port_id)
+            {
+                if (itr->first.mac == mac || mac == flush_mac)
+                {
+                    bool found = m_portsOrch->getPort(itr->first.port_name, port);
+                    port_temp = found ?  port : port_empty; 
+                    updates.emplace_back(itr->first.mac, itr->first.bv_id, fdb_type, port_temp, false);
+                }
+            }
+        }
+    }
+
+    for (auto& update : updates)
+    {
+        /* Fetch Vlan and decrement the counter */
+        Port temp_vlan;
+        if (m_portsOrch->getPort(update.entry.bv_id, temp_vlan))
+        {
+            temp_vlan.m_fdb_count--;
+            m_portsOrch->setPort(temp_vlan.m_alias, temp_vlan);
+        }
+
+        /* Decrement port fdb_counter */
+        Port temp_port;
+        if (m_portsOrch->getPort(update.port.m_alias, temp_port))
```

Comment:
Why do you need to get port here again when you already obtained it previously. You just need to skip the zero port alone here